src/main/java/net/datafaker/service/RandomService.java

```diff
@@ -141,6 +151,101 @@ public Random getRandomInternal() {
         return random;
     }
 
+    /**
+     * Returns a weighted random element from the given list, where each element is represented as a Map
+     * containing a weight and the corresponding value.
+     *
+     * @param items A list of maps, where each map contains:
+     *              - weight: A Double representing the weight of the element, influencing its selection probability.
+     *              - value: The actual element of type T to be randomly selected based on its weight.
+     * @param <T> The type of the element to be selected from the list. The value associated with the weight can be of any type.
+     * @return A randomly selected element based on its weight.
+     * @throws IllegalArgumentException if the list is null, empty, if any item in the list is null or empty,
+     *                                  if any weight is null, non-positive, NaN, negative zero, infinite,
+     *                                  or if the item does not contain 'weight' or 'value' keys.
+     */
+    public <T> T weightedArrayElement(List<Map<String, Object>> items) {
+        validateItemsList(items);
+        preprocessItems(items);
+        double randomValue = random.nextDouble() * cumulativeWeights[cumulativeWeights.length - 1];
+
+        return selectWeightedElement(randomValue);
+    }
+
+    private void validateItemsList(List<Map<String, Object>> items) {
+        if (items == null || items.isEmpty()) {
+            throw new IllegalArgumentException("weightedArrayElement expects a non-empty list");
+        }
+
+        Set<Object> uniqueValues = new HashSet<>();
+
+        for (var item : items) {
+            validateItem(item);
+            assertUniqueValues(item, uniqueValues);
+        }
+    }
+
+    private void assertUniqueValues(Map<String, Object> item, Set<Object> values) {
+        Object value = item.get(VALUE_KEY);
+        if (!values.add(value)) {
+            throw new IllegalArgumentException("Duplicate value found: " + value + ". Values must be unique.");
+        }
+    }
+
+    private void validateItem(Map<String, Object> item) {
+        if (item == null || item.isEmpty()) {
+            throw new IllegalArgumentException("Item cannot be null or empty");
+        }
+        if (!item.containsKey(WEIGHT_KEY) || !item.containsKey(VALUE_KEY)) {
```

Comment:
same here about equals/hashcode

Reply:
Sorry, but I might not get this. I am checking if each item contains weight and value keys defined as strings. If for some reason they do not, I am directly throwing an exception.